File: foodhut/customerService/src/main/java/com/stackroute/customerService/domain/Customer.java
package com.stackroute.customerService.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;


@Data
@AllArgsConstructor
@NoArgsConstructor
@Document
public class Customer {
    @Id
    private String emailId;
    private String username;
    private Address address;
    private List<Order> orderList;
    private Long contactno;
    private List<String> favourites;
    private List<String> favouriteProducts;
    private List<Item> cart;
    private Address deliveryAddress;
//    private List<String> favourites;
}

File: app/model/Customer.ts
import { Address } from "./Address";
import { Item } from "./Item";
import { Order } from "./Order";

export type Customer ={
    emailId?:string,
    username?:string,
    contactno?:string,
    address?:Address,
    orderList?:Order[],
    favourites?:string[],
    favouriteProducts?:string[],
    cart?:Item[],
    deliveryAddress?:Address
}